flow/Arena.h

```diff
@@ -429,18 +429,37 @@ class StringRef {
 		return StringRef(s,prefix.size() + size());
 	}
 
+	StringRef withSuffix( const StringRef& suffix, Arena& arena ) const {
+		uint8_t* s = new (arena) uint8_t[ suffix.size() + size() ];
+		memcpy(s, begin(), size());
+		memcpy(s+size(), suffix.begin(), suffix.size());
+		return StringRef(s,suffix.size() + size());
+	}
+
 	Standalone<StringRef> withPrefix( const StringRef& prefix ) const {
 		Standalone<StringRef> r;
 		((StringRef &)r) = withPrefix(prefix, r.arena());
 		return r;
 	}
 
+	Standalone<StringRef> withSuffix( const StringRef& suffix ) const {
+		Standalone<StringRef> r;
+		((StringRef &)r) = withSuffix(suffix, r.arena());
```

Comment:
`r.contents() = ` is a less questionable way to do this.

Reply:
Fixed. I was following the same convention as `withPrefix`, so I went and changed both of them.